src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/WrappedKeySecurityToken.cs

```diff
@@ -21,6 +21,38 @@ public class WrappedKeySecurityToken : SecurityToken
         bool serializeCarriedKeyName;
         byte[] wrappedKeyHash;
         XmlDictionaryString wrappingAlgorithmDictionaryString;
+        private byte[] unwrappedKey;
+        private string encryptionMethod;
+        private SecurityToken unwrappingToken;
+        private SecurityKeyIdentifier unwrappingTokenIdentifier;
+        private SecurityKey unwrappingSecurityKey;
+
+        internal WrappedKeySecurityToken(string id, byte[] keyToWrap, ISspiNegotiation wrappingSspiContext)
+        : this(id, keyToWrap, (wrappingSspiContext != null) ? (wrappingSspiContext.KeyEncryptionAlgorithm) : null, wrappingSspiContext, null)
+        {
+
+        }
+
+        internal WrappedKeySecurityToken(string id, byte[] keyToWrap, string wrappingAlgorithm, ISspiNegotiation wrappingSspiContext, byte[] wrappedKey)
+        : this(id, keyToWrap, wrappingAlgorithm, null)
+        {
+            if (wrappingSspiContext == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("wrappingSspiContext");
+            }
+            //this.wrappingSspiContext = wrappingSspiContext;
+            if (wrappedKey == null)
```

Comment:
I know it doesn't look like this is used, but I suspect this is held on to here to prevent the GC collecting it as it wraps a SafeHandle and running the finalizer on the safehandle will destroy the security context. I'm not 100% certain, but that's my suspicion.

Reply:
now that we are not using , I think it's ok to keep commented...

Reply:
If it's not needed any more, remove the code. I only want code left behind which will be brought back later.

---
In reply to: [565652789](https://github.com/CoreWCF/CoreWCF/pull/279#discussion_r565652789) [](ancestors = 565652789)